src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/YFlowCrudSpec.groovy

```diff
@@ -36,122 +51,498 @@ class YFlowCrudSpec extends HealthCheckSpecification {
-            withPool {
-                [exam.forward, exam.reverse].eachParallel { direction ->
+        def traffExam = traffExamProvider.get()
+        List<ExamReport> examReports
+        if (trafficApplicable) {
+            def exam = new FlowTrafficExamBuilder(topology, traffExam).buildYFlowExam(yFlow, yFlow.maximumBandwidth, 5)
+            examReports = withPool {
+                [exam.forward1, exam.reverse1, exam.forward2, exam.reverse2].collectParallel { Exam direction ->
                     def resources = traffExam.startExam(direction)
                     direction.setResources(resources)
-                    assert traffExam.waitExam(direction).hasTraffic()
+                    traffExam.waitExam(direction)
                 }
             }
-        } catch (FlowNotApplicableException e) {
-            //flow is not applicable for traff exam. That's fine, just inform
-            log.warn(e.message)
-            trafficApplicable = false
         }
 
-        and: "Flow writes stats"
-        if(trafficApplicable) { statsHelper.verifyFlowWritesStats(flow.flowId, beforeTraffic, true) }
+        then: "Traffic flows on both sub-flows, but does not exceed the y-flow bandwidth restriction (~halves for each sub-flow)"
+        if (trafficApplicable) {
+            def assertions = new SoftAssertions()
+            examReports.each { report ->
+                assertions.checkSucceeds {
+//                    def flowBwBits = yFlow.maximumBandwidth * 1000
+                    //allow 10% deviation
+//                    assert Math.abs(report.producerReport.bitsPerSecond - flowBwBits / 2) < flowBwBits * 0.1, report.exam
+                    assert report.hasTraffic(), report.exam
+                }
+            }
+            assertions.verify()
+        }
 
-        when: "Remove the flow"
-        flowHelperV2.deleteFlow(flow.flowId)
+        and: "Y-flow and subflows stats are available (flow.raw.bytes)"
+//        statsHelper.verifyFlowWritesStats(yFlow.YFlowId, beforeTraffic, trafficApplicable)
+        yFlow.subFlows.each {
+            statsHelper.verifyFlowWritesStats(it.flowId, beforeTraffic, trafficApplicable)
+        }
 
-        then: "The flow is not present in NB"
-        !northboundV2.getAllFlows().find { it.flowId == flow.flowId }
-        def flowIsDeleted = true
+        when: "Delete the y-flow"
+        northboundV2.deleteYFlow(yFlow.YFlowId)
 
-        and: "ISL bandwidth is restored"
-        Wrappers.wait(WAIT_OFFSET) {
-            def allLinksInfoAfter = northbound.getAllLinks().collectEntries { [it.id, it.availableBandwidth] }.sort()
-            assert allLinksInfoBefore == allLinksInfoAfter
+        then: "The y-flow is no longer visible via 'get' API"
+        Wrappers.wait(WAIT_OFFSET) { assert northboundV2.getAllYFlows().empty }
+        def flowRemoved = true
+
+        and: "Related sub-flows are removed"
+        Wrappers.wait(WAIT_OFFSET) { assert northboundV2.getAllFlows().empty }
```

Comment:
as for me, it can be unstable due to parallel execution. Probably would be better to check that sub-flowIds are not returned in the response.

Reply:
It's a special 'island' implementation of northboundV2 that will return flows that only belong to the current active topology of this thread. See `IslandNorthboundServiceV2 extends NorthboundServiceV2Impl`